doc/sphinxext/related_software.py

```diff
@@ -0,0 +1,226 @@
+    ) as url:
+        data = url.read().decode("utf-8")
+    # Parse data for list of names of packages
+    lines = [line.strip() for line in data.splitlines()]
+    start_idx = lines.index("# <<< BEGIN RELATED SOFTWARE LIST >>>") + 1
+    stop_idx = lines.index("# <<< END RELATED SOFTWARE LIST >>>")
+    packages = [
+        # Lines look like
+        # - mne-ari =0.0.0
+        # or similar.
+        line.split()[1]
+        for line in lines[start_idx:stop_idx]
+        if not line.startswith("#")
+    ]
+    return packages
+
+
+@functools.lru_cache
+def _get_packages():
+    packages = _get_installer_packages()
+    for name in MANUAL_PACKAGES:
+        if name not in packages:
+            packages.append(name)
+    # Simple alphabetical order
+    packages = sorted(packages, key=lambda x: x.lower())
+    packages = [RENAMES.get(package, package) for package in packages]
+    out = dict()
+    for package in status_iterator(
+        packages, f"Adding {len(packages)} related software packages: "
+    ):
+        out[package] = dict()
+        try:
+            if package in MANUAL_PACKAGES:
+                md = MANUAL_PACKAGES[package]
+            else:
+                md = importlib.metadata.metadata(package)
+        except importlib.metadata.PackageNotFoundError:
+            pass  # raise a complete error later
+        else:
+            # Every project should really have this
+            for key in ("Summary",):
+                if key not in md:
+                    raise ExtensionError(f"Missing {repr(key)} for {package}")
+            # It is annoying to find the home page
+            url = None
+            if "Home-page" in md:
+                url = md["Home-page"]
+            else:
+                for prefix in ("homepage", "documentation"):
+                    for key, val in md.items():
+                        if key == "Project-URL" and val.lower().startswith(
+                            f"{prefix}, "
+                        ):
+                            url = val.split(", ", 1)[1]
+                            break
+                    if url is not None:
+                        break
+                else:
+                    raise RuntimeError(
+                        f"Could not find Home-page for {package} in:\n"
+                        f"{sorted(set(md))}\nwith Summary:\n{md['Summary']}"
+                    )
+            out[package]["url"] = url
+            out[package]["description"] = md["Summary"].replace("\n", "")
+    bad = [package for package in packages if not out[package]]
+    if bad and REQUIRE_METADATA:
+        raise ExtensionError(f"Could not find metadata for:\n{' '.join(bad)}")
+
+    return out
+
+
+class RelatedSoftwareDirective(Directive):
+    """Create a directive that inserts a bullet list of related software."""
+
+    def run(self):
+        """Run the directive."""
+        # TODO: Consider a definition list maybe?
```

Comment:
```suggestion
```
I Don't think a definition list is helpful here. Styling as a simple list seems fine to me.